spec/classes/puppet_agent_install_spec.rb

```diff
@@ -21,14 +32,16 @@
     end
 
     context "on a Windows family OS" do
-      let :facts do {
-        :osfamily => 'windows',
-        :concat_basedir => 'C:\Temp',
-        :puppetversion => Puppet.version,
-      } end
+      let :facts do
+        {
+          :osfamily => 'windows',
+          :concat_basedir => 'C:\Temp',
+          :puppetversion => Puppet.version,
+        }.merge(additional_facts)
+      end
 
       it 'should define provider as chocolatey' do
-        should contain_package('puppet').with_provider('chocolatey')
+        should contain_package(client_package).with_provider('chocolatey')
```

Comment:
While testing the MSI-based install I found a couple of bugs not directly related to these changes. I do we want to fix them prior to merging this PR or included them in their own PR... I am thinking the latter.